### PR TITLE
RFC: Build LLVM as a shared library by default

### LINKED ISSUE #16117: Build LLVM as shared library by default
In order for Cxx.jl to be usable on a default julia distribution we need to either ship LLVM as a shared library or re-export the LLVM symbols. Given historic opposition to doing the latter, I suggest doing the former.

### PR DESCRIPTION
**Pull request by tkelman**
fixes #16117

needs testing on CI and verification on buildbots

### CODE REVIEW

**Comment by tkelman** [comment]
Looks like I'll need to fix llvm's build system to stop it from making cygwin symlinks for `LLVM-3.7.1.dll`.

File: appveyor.yml [hunk 1]
```diff
@@ -29,11 +29,11 @@ init:
...
 
 cache:
-# Cache any downloads from sourceforge because it's really unreliable
+# Cache large downloads to avoid network unreliability
   - i686-4.9.2-release-win32-sjlj-rt_v4-rev3.7z
   - x86_64-4.9.2-release-win32-seh-rt_v4-rev3.7z
-  - llvm-3.7.1-i686-w64-mingw32-juliadeps-r04.7z
-  - llvm-3.7.1-x86_64-w64-mingw32-juliadeps-r04.7z
+  - llvm-3.7.1-i686-w64-mingw32-juliadeps-r09.7z
+  - llvm-3.7.1-x86_64-w64-mingw32-juliadeps-r09.7z
```
Comment (tkelman) [inline appveyor.yml]: Note, mostly to myself, that the separate llvm builds/downloads probably won't be necessary any more after this is merged and makes it into the binaries, at least we'll be able to get the dll's from the nightlies. Though we'd need to still get the (post-configure, I think) includes from somewhere, so maybe I'm coming around a bit on #16093.

**Comment by tkelman** [comment]
I guess we're in business here, that wasn't so bad. A few manual builds seem to have gone through okay on the buildbots, but I haven't done much testing of them.

**Comment by yuyichao** [comment]
This seems to have break gdb backtrace on Linux?

**Comment by tkelman** [comment]
@yuyichao did that ever get fixed?

**Comment by yuyichao** [comment]
no

**Pull request merged**